docs/user-management/user-data.mdx

```diff
@@ -291,6 +307,82 @@ then new _custom_data_ value after updating should be:
 
 That is, the updated field value has nothing to do with the previous value.
 
+## Roles \{#roles}
+
+_roles_ is an array of roles assigned to the user for [role-based access control](/authorization/role-based-access-control) (RBAC). A user can have one or more roles, with each role stored as an individual JSON object. Each role typically includes basic role information, and the [API resources](/authorization/api-resources) and [scopes (permissions)](/authorization/role-based-access-control/configure-permissions) the user can access. Here’s a copy of the type definition:
+
+```tsx
+roles: {
+  id: string;
+  name: string;
+  description: string;
+  scopes: {
+    id: string;
+    name: string;
+    description: string | null;
+    resourceId: string;
+    resource: {
+      tenantId: string;
+      id: string;
+      name: string;
+      indicator: string;
+      isDefault: boolean;
+      accessTokenTtl: number;
+    }
+  }
+  [];
+}
+[];
```

Comment:
Hmmm, would prefer using `Array<{ key: type; key2: type }>` when defining the types for a complex object, instead of `[]`